library/include/Core.h

```diff
@@ -197,13 +203,7 @@ namespace DFHack
         struct Private;
         std::unique_ptr<Private> d;
 
-        bool Init();
-        int Update (void);
-        int Shutdown (void);
-        int DFH_SDL_Event(SDL::Event* event);
-        bool ncurses_wgetch(int in, int & out);
-
-        void doUpdate(color_ostream &out, bool first_update);
```

Comment:
Access to these functions was previously controlled via the `friend` declarations at the top of this class. It could be worth updating those declarations instead of making these public. Or if we really need to make them public, we can remove the `friend`s.

Reply:
These functions are now being called from another library (libdfhooks.so), so I don't know if `friend` is the right mechanism anymore. We could probably find a way to keep them from being public if that makes the API cleaner though..

Reply:
Isn't `library/Hooks.cpp` part of our own library? That's the only place that directly calls the Core methods that I can see. Really, `friend` keeps things like plugins from calling `Core::Init()`.

Reply:
Only in the windows build. On Linux and osx it's compiled into a separate "dfhooks" library.

We could have yet another bounce point that could befriend Core and keep its functions private, though

Reply:
Oh I see. Well, `friend` is only enforced by the compiler, not the linker, similar to private/public, so `friend` should still be capable of allowing functions to access private members even if those functions are in another translation unit, shared object, etc.

Reply:
rewritten to be much cleaner. Thanks!

Reply:
notably, we don't actually need to pull in the DFHack/hooks code as a submodule. The header file there is for the DF side of the API, not ours.